### PR TITLE
Add nodeSelector and tolerations to chart

### PR DESCRIPTION
**Pull request by cmotta2016**
#### What does this implement/fix?
This MR implements the ability to use nodeSelector and tolerations for scheduling pods on specific nodes.

#### Why do we need it?
In some scenarios, PODs must be scheduled at specific nodes and evict other ones.

### CODE REVIEW

File: charts/pvc-autoresizer/templates/controller/deployment.yaml [hunk 1]
```diff
@@ -51,3 +51,11 @@ spec:
...
               path: /healthz
               port: health
+    {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
```
Comment (toshipp) [inline charts/pvc-autoresizer/templates/controller/deployment.yaml:61]: Would you add a new line?

File: charts/pvc-autoresizer/values.yaml [hunk 2]
```diff
@@ -33,3 +33,9 @@ controller:
 
   # controller.terminationGracePeriodSeconds -- Specify terminationGracePeriodSeconds.
   terminationGracePeriodSeconds:  # 10
+
+# tolerations -- Ensure pods are not scheduled on inappropriate nodes
+tolerations: {}
+
+# nodeSelector -- Map of key-value pairs for scheduling pods on specific nodes
+nodeSelector: {}
```
Comment (toshipp) [inline charts/pvc-autoresizer/values.yaml:41]: We currently have only one controller deployment, however, there would be more deployments in the future.
Therefore, these maps should be under the controller key, IMO

**Review by toshipp — CHANGES_REQUESTED** [review]

File: charts/pvc-autoresizer/templates/controller/deployment.yaml (hunk 1 above)
Comment (cmotta2016) [reply charts/pvc-autoresizer/templates/controller/deployment.yaml]: Done!

File: charts/pvc-autoresizer/values.yaml (hunk 2 above)
Comment (cmotta2016) [reply charts/pvc-autoresizer/values.yaml]: Done!

File: charts/pvc-autoresizer/values.yaml [hunk 3]
```diff
@@ -33,3 +33,9 @@ controller:
 
   # controller.terminationGracePeriodSeconds -- Specify terminationGracePeriodSeconds.
   terminationGracePeriodSeconds:  # 10
+
+  # controller.tolerations -- Ensure pods are not scheduled on inappropriate nodes
```
Comment (toshipp) [inline charts/pvc-autoresizer/values.yaml:37]:
```suggestion
  # controller.tolerations -- Ensure pods are not scheduled on inappropriate nodes.
```

File: charts/pvc-autoresizer/values.yaml [hunk 4]
```diff
@@ -33,3 +33,9 @@ controller:
...
+  tolerations: {}
+  
+  # controller.nodeSelector -- Map of key-value pairs for scheduling pods on specific nodes
```
Comment (toshipp) [inline charts/pvc-autoresizer/values.yaml:40]:
```suggestion
  # controller.nodeSelector -- Map of key-value pairs for scheduling pods on specific nodes.
```

**Review by toshipp — CHANGES_REQUESTED** [review]
helm-docs generates README.me from values.yaml.
I suggested appropriate changes based on local helm-docs result.

```
diff --git a/charts/pvc-autoresizer/README.md b/charts/pvc-autoresizer/README.md
index 10b79cb..ce0f986 100644
--- a/charts/pvc-autoresizer/README.md
+++ b/charts/pvc-autoresizer/README.md
@@ -34,11 +34,11 @@ helm upgrade --create-namespace --namespace pvc-autoresizer -i pvc-autoresizer -
 | controller.args.additionalArgs | list | `[]` | Specify additional args. |
 | controller.args.interval | string | `"10s"` | Specify interval to monitor pvc capacity. Used as "--interval" option |
 | controller.args.prometheusURL | string | `"http://prometheus-prometheus-oper-prometheus.prometheus.svc:9090"` | Specify Prometheus URL to query volume stats. Used as "--prometheus-url" option |
-| controller.nodeSelector | object | `{}` | Map of key-value pairs for scheduling pods on specific nodes. |
+| controller.nodeSelector | object | `{}` | Map of key-value pairs for scheduling pods on specific nodes |
 | controller.replicas | int | `1` | Specify the number of replicas of the controller Pod. |
 | controller.resources | object | `{"requests":{"cpu":"100m","memory":"20Mi"}}` | Specify resources. |
 | controller.terminationGracePeriodSeconds | string | `nil` | Specify terminationGracePeriodSeconds. |
-| controller.tolerations | object | `{}` | Ensure pods are not scheduled on inappropriate nodes. |
+| controller.tolerations | object | `{}` | Ensure pods are not scheduled on inappropriate nodes |
 | image.pullPolicy | string | `nil` | pvc-autoresizer image pullPolicy. |
 | image.repository | string | `"quay.io/topolvm/pvc-autoresizer"` | pvc-autoresizer image repository to use. |
 | image.tag | string | `{{ .Chart.AppVersion }}` | pvc-autoresizer image tag to use. |
```

**Comment by toshipp** [comment]
Need #70 to resolve chart lint failing.

**Review by cmotta2016 — COMMENTED** [review]
@toshipp hey man, can you revert [this](https://github.com/topolvm/pvc-autoresizer/pull/69/commits/3ba01840ea1ba176053654f8ddbeae39f84fab66) commit, please? I didn`t see that another PR will change this, my bad.

**Comment by toshipp** [comment]
@cmotta2016
Yeah, I can. BTW you can do that without my help.
Try it as following.
```sh
$ git reset HEAD^^
$ git log    # HEAD is now 2a4f5ba
$ git push -f origin chart-add-nodeselector-and-tolerations 
```
If you face any trouble, I will do this.

**Comment by toshipp** [comment]
Ah, I just realized that the PR is from your repository, so I can't...
You should do this yourself.

**Comment by cmotta2016** [comment]
> @cmotta2016 Yeah, I can. BTW you can do that without my help. Try it as following.
> 
> ```shell
> $ git reset HEAD^^
> $ git log    # HEAD is now 2a4f5ba
> $ git push -f origin chart-add-nodeselector-and-tolerations 
> ```
> 
> If you face any trouble, I will do this.

It works. Very thanks, bro.

**Comment by toshipp** [comment]
@cmotta2016 
We've merged #70.
Would you rebase your branch?
Here is a procedure.
```sh
$ git fetch upstream # change `upstream` if you use another remote name
$ git rebase upstream/main
$ git push -f
```

**Comment by cmotta2016** [comment]
> @cmotta2016 We've merged #70. Would you rebase your branch? Here is a procedure.
> 
> ```shell
> $ git fetch upstream # change `upstream` if you use another remote name
> $ git rebase upstream/main
> $ git push -f
> ```

Done!

**Review by toshipp — APPROVED** [review]
LGTM

**Review by cupnes — APPROVED** [review]
LGTM

**Pull request merged**